### PR TITLE
Q91 Add decodeWays.py

### PR DESCRIPTION
**Pull request by peinanteng**

### CODE REVIEW

File: decodeWays.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    def numDecodings(self, s):
+        """
+        :type s: str
+        :rtype: int
+        """
+        if s == '' or s[0] == '0':
+            return 0
+        if len(s) == 1:
+            return 1
+        prePreCount, preCount, curCount = 1, 1, 0
```
Comment (nkukarl) [inline decodeWays.py]: `prePreCount`... what a mysterious name...

File: decodeWays.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        for i in range(1, len(s)):
+            flag1 = self.decodeTwoDigit(s[i - 1: i + 1])
+            flag2 = self.decodeDigit(s[i])
+            if flag1 == 0 and flag2 == 0 :
```
Comment (nkukarl) [inline decodeWays.py]: `flag1` and `flag2` shall be returned as boolean values...
Maybe should do something like
```
if not flag1 and not flag2
```

File: decodeWays.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+            elif flag2 == 0:
+                curCount = prePreCount
+            else:
+                curCount = preCount + prePreCount
+            prePreCount, preCount = preCount, curCount
+        return curCount
+    
+    def decodeDigit(self, digit):
+        return digit != '0'
+    def decodeTwoDigit(self, digit):
```
Comment (nkukarl) [inline decodeWays.py]: Blank line

File: decodeWays.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        return digit[0] == '1' or (digit[0] == '2' and digit[1] <= '6')
```
Comment (nkukarl) [inline decodeWays.py]: Maybe add some comments

File: decodeWays.py [hunk 5]
```diff
@@ -12,18 +12,21 @@ def numDecodings(self, s):
         for i in range(1, len(s)):
             flag1 = self.decodeTwoDigit(s[i - 1: i + 1])
             flag2 = self.decodeDigit(s[i])
-            if flag1 == 0 and flag2 == 0 :
+            if not flag1 and not flag2:
                 return 0
-            elif flag1 == 0:
+            elif not flag1:
```
Comment (nkukarl) [inline decodeWays.py]: Actually, this should be `if` not `elif`.

Comment (peinanteng) [reply decodeWays.py]: I tried, this should be elif.

**Pull request merged**